File: day1/2.cpp
#include <fstream>
#include <iostream>
#include <string>

int calc_fuel(int fuel){
    int ans = 0;
    while (fuel > 0){
        fuel = fuel / 3 - 2;
        if (fuel > 0) ans += fuel;
    }
    return ans;
}

int main() {
    std::ifstream in("data.txt"); std::string line;
    int ans{};
    while (std::getline(in, line)){ 
        int init = stoi(line);
        ans += calc_fuel(init);
        }
    in.close();
    std::cout << ans;
}

File: day1/2.py
def calc_fuel(init: int) -> int:
    sum = 0
    while (init > 0):
        init = init // 3 - 2
        if (init > 0): sum += init
    return sum

ans = 0

with open("data.txt", 'r') as file:
    for line in file:
        ans += calc_fuel(int(line))

print(ans)